java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -5312,4 +5313,100 @@ public void lcs_with_len_option(BaseClient client) {
                 assertThrows(ExecutionException.class, () -> client.lcs(nonStringKey, key1).get());
         assertInstanceOf(RequestException.class, executionException.getCause());
     }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void test_dump_restore(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        String newKey = UUID.randomUUID().toString();
+        String tempKey = UUID.randomUUID().toString();
+        String nonExistingKey = UUID.randomUUID().toString();
+        String value = "oranges";
+
+        assertEquals(OK, client.set(key, value).get());
+
+        // Dump existed key
+        byte[] result = client.dump(key.getBytes()).get();
+        assertNotNull(result.toString());
+
+        // Dump non-existing key
+        assertNull(client.dump(nonExistingKey.getBytes()).get());
+
+        // Restore to a new key
+        assertEquals(OK, client.restore(newKey.getBytes(), 0L, result).get());
+
+        // Restore to an existed key - Error: "Target key name already exists"
+        Exception executionException =
+                assertThrows(
+                        ExecutionException.class, () -> client.restore(newKey.getBytes(), 0L, result).get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+
+        // Restore with checksum error - Error: "payload version or checksum are wrong"
+        executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () -> client.restore(tempKey.getBytes(), 0L, value.getBytes()).get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
```

Comment:
should probably include a test for when the supplied key doesn't have the right typing

Reply:
When `Replace` option is set, Restore command will replace the key regardless its previous type.